### ISSUE TITLE
Incorrect announcement of Regions

### ISSUE THREAD

**Issue by ptorpey** [jcsteh/osara#442]
The semicolon and apostrophe keys move from marker to marker and also move from region to region.  When regions overlap, these keys provide incorrect feedback of the region name as well as the start and end points of the region.  Here is an example:

1.  Set edit markers at the beginnings of measures 10 and 20 using the left and right brackets.
2.  Hit Shift+R to create a region from the selected range.
3.  Hit shift+alt+R to label this "Region #2".
4.  Set edit markers at the beginnings of measures 1 and 15 using the left and right brackets.
5. Create a region by hitting shift+R.
6. Hit shift+alt+R to label the region "Region #1
7. Hit the semicolon and apostrophe keys to navigate through the markers associated with the beginnings and ends of the regions.  Here is what you will find in order::
measure 1: "start of region #1"
measure 10: "start of region #2"
measure 15: "end of region #2"
measure 20: "end of project"
Note: measure 15 is incorrectly labeled as the end of region #2
9.  You can verify the start and end of each region by hitting control+shift+alt+R to get into the Markers and Region Manager.  The start and end of each region is properly labeled in this dialog.
10. Now use the semicolon and apostrophe keys to move to measure 15.  This is reported incorrectly as the end of region #2.
11. Hit delete.
12. Instead of deleting region #2 you will find that region #1 is deleted.
Summary: The navigation commands are providing incorrect info for the region markers which can lead to errors in deleting the desired region.
10. 9.

**Comment by jcsteh** [comment]
OSARA doesn't provide feedback about the end of regions. This is probably a bug - I didn't realise those commands moved to the region ends as well - but I'm somewhat confused by your report saying that you do hear about region ends. I wonder if this is a function of the JAWS scripts?

**Comment by jcsteh** [comment]
To clarify, OSARA reports "blah region", where blah is the region's name, or "region 1", where 1 is the number. However, it doesn't report "start of region" or "end of region".

**Comment by jcsteh** [comment]
That said, I can still trigger bugs where OSARA reports the wrong region at the end position if there are overlapping regions.

Fixing this is going to be a bit tricky because REAPER has an API call to retrieve the most recent region for a given time, but for the end position where there are overlapping regions, that's not necessarily the region we want.

**Comment by jcsteh** [comment]
> 10. Hit delete.
> 11. Instead of deleting region #2 you will find that region #1 is deleted.

When you press delete here, OSARA just uses REAPER's "Markers: Delete region near cursor" command. That suggests REAPER itself doesn't know how to handle the end positions of regions in the presence of overlapping regions. So, even if we fix OSARA's reporting, pressing delete is still going to delete the wrong region.

**Comment by ScottChesworth** [comment]
One for Cockos then, @jcsteh?

**Comment by ptorpey** [comment]
Sorry my report was misleading.  Actually you are correct, Osara is not reporting the ends of regions.  I was inferring the end of a region when the apostrophe key moved me to the end of a region where I knew that I didn’t have any markers.  Also, in general, my regions were adjacent to each other so that when I navigated to the end of one region I was placed at the beginning of the next region and the name of that next region was spoken.  But for where there was no adjacent next region, when I moved to the end of a region Osara just spoke the time.

 

--pete
